cmd/cluster-agent/api/v1/clusterchecks.go

```diff
@@ -159,13 +163,14 @@ func clusterChecksDisabledHandler(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte("Cluster-checks are not enabled"))
 }
 
-// parseClientIP retrieves the http client IP from the remoteAddr attribute
-func parseClientIP(remoteAddr string) (string, error) {
-	clientIP, _, err := net.SplitHostPort(remoteAddr)
-	if err != nil {
-		log.Debugf("Error while parsing CLC worker address %s: %v", remoteAddr, err)
-		return "", err
+// validateClientIP validates the http client IP retrieved from the request's header.
+// Empty IPs are considered valid for backward compatibility with old clc runner versions
+// that don't set the clcRunnerIpHeader header field.
+func validateClientIP(addr string) (string, error) {
+	if addr != "" && net.ParseIP(addr) == nil {
```

Comment:
This means for old agents that don t send the header, it won't work, but it won't error out right? Is that the behavior we want? Do we want to log a warning that suggests to upgrade agents if the address is empty maybe?